lib/openhab/dsl/monkey_patch/items/persistence.rb

```diff
@@ -65,14 +65,26 @@ def previous_state(service = nil, skip_equal: false)
           #
           # @return [ZonedDateTime]
           #
+          # rubocop: disable Metrics/AbcSize
+          # rubocop: disable Metrics/MethodLength
           def to_zdt(timestamp)
-            if timestamp.is_a? Java::JavaTimeTemporal::TemporalAmount
-              logger.trace("Converting #{timestamp} (#{timestamp.class}) to ZonedDateTime")
-              Java::JavaTime::ZonedDateTime.now.minus(timestamp)
-            else
-              timestamp
+            java_import Java::JavaTime::ZonedDateTime
+            logger.trace("Converting #{timestamp} (#{timestamp.class}) to ZonedDateTime")
+            case timestamp
+            when ZonedDateTime then timestamp
+            when Java::JavaTimeTemporal::TemporalAmount then ZonedDateTime.now.minus(timestamp)
+            when OpenHAB::DSL::TimeOfDay::TimeOfDay
+              ZonedDateTime.now
+                           .withHour(timestamp.hour)
+                           .withMinute(timestamp.minute)
+                           .withSecond(timestamp.second)
+                           .withNano(0)
+            when String then to_zdt(OpenHAB::DSL::TimeOfDay::TimeOfDay.parse(timestamp))
```

Comment:
It might make sense to add a to_zdt method to TimeOfDay, then you could change to:
```ruby
when OpenHAB::DSL::TimeOfDay::TimeOfDay then datetime.to_zdt
            when String then OpenHAB::DSL::TimeOfDay::TimeOfDay.parse(timestamp).to_zdt
```
It would then be easy to also support passing a DateTimeItem as parameter by adding
```ruby
return datetime.to_zdt if datetime.respond_to?(:to_zdt)
```
As a guard clause at the top (and remove `when TimeOfDay)

Reply:
Great idea!